include/boost/url/detail/params_iterator_impl.hpp

```diff
@@ -27,23 +27,20 @@ struct params_iterator_impl
     std::size_t nk_ = 0;
     std::size_t nv_ = 0;
     char const* end_ = nullptr;
-    const_string::factory a_;
 
     BOOST_URL_DECL
     void scan() noexcept;
 
     BOOST_URL_DECL
     params_iterator_impl(
-        string_view s,
-        const_string::factory a) noexcept;
+        string_view s) noexcept;
 
     // end ctor
     BOOST_URL_DECL
     params_iterator_impl(
         string_view s,
         std::size_t nparam,
-        int,
-        const_string::factory a) noexcept;
+        int) noexcept;
```

Comment:
This is confusing. If you have `nparam` why do you also need the dummy `int`? The begin constructor takes only `string_view`.